crates/underwriter/src/lookahead_fetcher.rs

```diff
@@ -53,37 +56,60 @@ impl LookaheadFetcher {
         self.network_state.update_slot(slot);
 
         // Update the fee recipients for the current epoch
-        let fee_recipients = self.relay_client.get_current_epoch_validators().await?;
+        let fee_recipients = self.get_current_epoch_validators().await?;
         self.network_state.update_fee_recipients(fee_recipients);
         Ok(())
     }
 
+    pub async fn get_current_epoch_validators(&self) -> Result<Vec<ValidatorSlotData>> {
+        let mut validator_data = Vec::new();
+        let mut known_slots = Vec::new();
+        for relay_client in self.relay_clients.iter() {
```

Comment:
I am thinking maybe we should just look at only one relay here because they are all the same. Is it possible the epoch number is different that I request the first relay and then the second?